templates/Tutorial.md

```diff
@@ -21,3 +21,45 @@ uint32 "UInt32"
 4. Open any file and change the selection cursor to new locations. You'll see the "UInt32" field update.
 
 Congrats, you've written the most basic and simplest template that actually does something! Visit the [reference](Reference.md) for details on other commands available.
+
+## Development and debug tips
+
+### How to see debug logs
+
+Run application directly in a terminal:
+```sh
+$ /Applications/Hex\ Fiend.app/Contents/MacOS/Hex\ Fiend
+```
+Now in a template do `puts test` and you will see `test` on the terminal.
+
+### Keep partial template tree on error
+
+Wrap parse code in a `catch` and print error on crash. This will hide the error from Hex Fiend and show the partial tree.
+
+```tcl
+if [catch {
+    # call parse code
+}] {
+    puts $errorInfo
+}
+```
+
+### Automatically reload template on save
```

Comment:
we should eventually just support this natively.

Reply:
That would be great. Re-select/-expand on reload would be nice also but maybe hard as names not unique? some kind of best effort?